default/scripting/buildings/TERRAFORM.focs.txt

```diff
@@ -16,20 +15,13 @@ BuildingType
             Planet environment = Adequate
         ]
     ]
-    EnqueueLocation = And [
-        Planet
-        OwnedBy empire = Source.Owner
-        Not Contains Building name = "BLD_TERRAFORM_REVERT"
-        Not Contains Building name = "BLD_GAIA_TRANS"
-        Not HasSpecial name = "GAIA_SPECIAL"
-        Not Enqueued type = Building name = "BLD_TERRAFORM_REVERT"
-        Not Enqueued type = Building name = "BLD_GAIA_TRANS"
-    ]
```

Comment:
I'm probably missing something important here, but why remove the `EnqueueLocation` clause entirely? I can see where you have to remove the `Not ... Building name = "BLD_TERRAFORM_REVERT"` parts, but shouldn't enqueuing a Terraformer building be prevented on planets with the Gaia special/building (present or enqueued)?

Moreover, with the proposed changes to the Terraformer building (one building doing all the terraforming required to achieve "good" environment), shouldn't enqueuing more than one of them at the same location be prevented?

Reply:
IIRC, `EnqueueLocation` inherits from `location` when omitted. This does need re-added though as you are correct, this building should not be enqueue-able a second time.